test/apivalidations/network_test.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2024 The ORC Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apivalidations
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	orcv1alpha1 "github.com/k-orc/openstack-resource-controller/api/v1alpha1"
+	applyconfigv1alpha1 "github.com/k-orc/openstack-resource-controller/pkg/clients/applyconfiguration/api/v1alpha1"
+)
+
+const (
+	networkName = "sg-foo"
+	networkID   = "365c9e4f-0f5a-46e4-9f3f-fb8de25ae12f"
+)
+
+func networkStub(namespace *corev1.Namespace) *orcv1alpha1.Network {
+	obj := &orcv1alpha1.Network{}
+	obj.Name = networkName
+	obj.Namespace = namespace.Name
+	return obj
+}
+
+func baseNetworkPatch(network client.Object) *applyconfigv1alpha1.NetworkApplyConfiguration {
+	return applyconfigv1alpha1.Network(network.GetName(), network.GetNamespace()).
+		WithSpec(applyconfigv1alpha1.NetworkSpec().
+			WithCloudCredentialsRef(testCredentials()))
+}
+
+var _ = Describe("ORC Network API validations", func() {
+	var namespace *corev1.Namespace
+	BeforeEach(func() {
+		namespace = createNamespace()
+	})
+
+	It("should allow to create a minimal network and managementPolicy should default to managed", func(ctx context.Context) {
+		network := networkStub(namespace)
+		patch := baseNetworkPatch(network)
+		patch.Spec.WithResource(applyconfigv1alpha1.NetworkResourceSpec())
+		Expect(applyObj(ctx, network, patch)).To(Succeed())
+		Expect(network.Spec.ManagementPolicy).To(Equal(orcv1alpha1.ManagementPolicyManaged))
+	})
+
+	It("should reject a network with empty name", func(ctx context.Context) {
+		network := networkStub(namespace)
+		network.Name = ""
+		patch := baseNetworkPatch(network)
+		patch.Spec.WithResource(applyconfigv1alpha1.NetworkResourceSpec())
+		Expect(applyObj(ctx, network, patch)).To(MatchError(ContainSubstring("resource name may not be empty")))
+	})
+
+	It("should reject a network with a comma in the network name", func(ctx context.Context) {
+		network := networkStub(namespace)
+		network.Name = "foo,bar"
+		patch := baseNetworkPatch(network)
+		patch.Spec.WithResource(applyconfigv1alpha1.NetworkResourceSpec())
+		Expect(applyObj(ctx, network, patch)).NotTo(Succeed())
+	})
```

Comment:
Same as above. Do you want to add this validation to spec.resource.name instead? This is just testing k8s own object validation of the name.

Reply:
I added
```
// +kubebuilder:validation:Pattern:="^[^,]+$"
type OpenStackName string
```
Do you still want to remove it?